api/src/test/java/org/openmrs/module/queue/utils/QueueUtilsTest.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.queue.utils;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+import java.util.Date;
+
+import org.junit.Test;
+
+public class QueueUtilsTest {
+	
+	private static final Date NULL = null;
+	
+	private static final Date AUG_1 = QueueUtils.parseDate("2023-08-01 10:00:00");
+	
+	private static final Date AUG_2 = QueueUtils.parseDate("2023-08-02 10:00:00");
+	
+	private static final Date AUG_3 = QueueUtils.parseDate("2023-08-03 10:00:00");
+	
+	private static final Date AUG_4 = QueueUtils.parseDate("2023-08-04 10:00:00");
+	
+	@Test
+	public void shouldReturnTrueIfDatesOverlap() {
+		// Test that nulls are handled as open-ended dates
```

Comment:
@cioan - see how I rewrote this test.  I found this style a lot easier for me to follow, and more conducive to adding lots of test criteria.  Have a look and let me know if you think any test cases are missing, and if you have questions about whether certain conditions are handled correctly.